crates/walrus-service/src/node.rs

```diff
@@ -2204,10 +2241,170 @@ impl StorageNode {
 }
 
 impl StorageNodeInner {
+    /// Computes a deferral duration from the configured policy and a given unencoded size.
+    pub fn deferral_for_unencoded_size(&self, size_bytes: u64) -> Option<std::time::Duration> {
+        self.live_upload_deferral_config
+            .deferral_for_size(size_bytes)
+    }
+
+    /// Returns the latest checkpoint sequence number known to the node.
+    pub fn latest_checkpoint_sequence_number(&self) -> Option<u64> {
+        self.event_manager.latest_checkpoint_sequence_number()
+    }
+
+    /// Applies a recovery deferral if the node determines the blob is part of an in-flight upload.
+    pub async fn maybe_apply_live_upload_deferral(
+        &self,
+        blob_id: BlobId,
+        checkpoint_sequence: u64,
```

Comment:
Question: What is the `checkpoint_sequence` here?

Reply:
added comment down below.